packages/core/src/baseclient.ts

```diff
@@ -658,6 +659,23 @@ export abstract class BaseClient<O extends ClientOptions> implements Client<O> {
           this._updateSessionFromEvent(session, processedEvent);
         }
 
+        const transactionInfo = processedEvent.transaction_info;
+        if (isTransaction && transactionInfo && processedEvent.transaction !== event.transaction) {
+          const source = 'custom';
```

Comment:
When I first saw this, my gut reaction was to worry that it was going to break the `RequestData` integration, until I realized that the integration shouldn't ever change a transaction event's `transaction`, only an error event's `transaction`. So we're good there, but still, I wonder if it might not be worth putting a note here basically saying "we know none of our stuff does this, so if it happened, it must be because of some custom event processor added by the user."